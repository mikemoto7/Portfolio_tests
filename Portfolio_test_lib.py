
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







