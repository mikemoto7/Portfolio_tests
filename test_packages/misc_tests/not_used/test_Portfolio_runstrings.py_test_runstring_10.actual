
Processing: user_runstring = ./Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml

Processing: --price_web_update

Processing:  --pt show_percent

---------------------------------------------------------------------------------------------------------
                          Tot       Tot     Tot        Ret       Ret    Ret    Non_Ret   Non_Ret  Non_Ret
Asset_Class         inv_count       amt     pct  inv_count       amt    pct  inv_count       amt      pct
Energy                      4   93775.4   10.68          3   74705.4  14.11          1   19070.0     5.46
Global                      6   59665.1    6.79          4   50127.4   9.47          2    9537.7     2.73
Healthcare                  2   26734.4    3.04          1    4312.0   0.81          1   22422.4     6.43
Individual_Stocks           1   28693.8    3.27          0       0.0   0.00          1   28693.8     8.22
Large_Blend                 3   45011.4    5.13          2   23990.4   4.53          1   21021.0     6.02
Large_Growth                4  116004.2   13.21          2   83871.6  15.85          2   32132.6     9.21
Large_Value                 3   48812.1    5.56          2   33788.7   6.38          1   15023.4     4.31
Mid_Blend_Aggress           3   75099.3    8.55          0       0.0   0.00          3   75099.3    21.52
Mid_Growth                  5   91000.3   10.36          3   73643.3  13.91          2   17357.0     4.97
Mid_Value                   5   55137.0    6.28          3   31159.0   5.89          2   23978.0     6.87
REIT                        5  110356.0   12.57          4   77084.3  14.56          1   33271.7     9.53
Small_Blend                 2   23400.7    2.66          1    4493.3   0.85          1   18907.4     5.42
Small_Growth                4   84902.2    9.67          3   69216.2  13.08          1   15686.0     4.50
Small_Value                 2   19668.8    2.24          1    2920.0   0.55          1   16748.8     4.80
Totals                     49  878260.7  100.01         29  529311.6  99.99         20  348949.1    99.99
                                                                                                         

Processing:  --pt lp

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------
                    Sorted                                               Sorted                                               Sorted
Asset                Tot_%  Ret_%  Non_%     Asset                Tot_%   Ret_%  Non_%     Asset                Tot_%  Ret_%   Non_%
Small_Value           2.24   0.55   4.80     Mid_Blend_Aggress     8.55    0.00  21.52     Global                6.79   9.47    2.73
Small_Blend           2.66   0.85   5.42     Small_Value           2.24    0.55   4.80     Large_Value           5.56   6.38    4.31
Healthcare            3.04   0.81   6.43     Healthcare            3.04    0.81   6.43     Small_Growth          9.67  13.08    4.50
Large_Blend           5.13   4.53   6.02     Small_Blend           2.66    0.85   5.42     Small_Value           2.24   0.55    4.80
Large_Value           5.56   6.38   4.31     Large_Blend           5.13    4.53   6.02     Mid_Growth           10.36  13.91    4.97
Mid_Value             6.28   5.89   6.87     Mid_Value             6.28    5.89   6.87     Small_Blend           2.66   0.85    5.42
Global                6.79   9.47   2.73     Large_Value           5.56    6.38   4.31     Energy               10.68  14.11    5.46
Mid_Blend_Aggress     8.55   0.00  21.52     Global                6.79    9.47   2.73     Large_Blend           5.13   4.53    6.02
Small_Growth          9.67  13.08   4.50     Small_Growth          9.67   13.08   4.50     Healthcare            3.04   0.81    6.43
Mid_Growth           10.36  13.91   4.97     Mid_Growth           10.36   13.91   4.97     Mid_Value             6.28   5.89    6.87
Energy               10.68  14.11   5.46     Energy               10.68   14.11   5.46     Large_Growth         13.21  15.85    9.21
REIT                 12.57  14.56   9.53     REIT                 12.57   14.56   9.53     REIT                 12.57  14.56    9.53
Large_Growth         13.21  15.85   9.21     Large_Growth         13.21   15.85   9.21     Mid_Blend_Aggress     8.55   0.00   21.52
Individual_Stocks     3.27   0.00   8.22     Individual_Stocks     3.27    0.00   8.22     Individual_Stocks     3.27   0.00    8.22
Totals              100.01  99.99  99.99     Totals              100.01   99.99  99.99     Totals              100.01  99.99   99.99

Processing:  --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing: --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing:  --pt show_percent

----------------------------------------------------------------------------------------------------------------------------------
                          Tot               Tot          Tot        Ret        Ret    Ret    Non_Ret           Non_Ret     Non_Ret
Asset_Class         inv_count               amt          pct  inv_count        amt    pct  inv_count               amt         pct
Energy *                    5  15000.0+93775.04  11.06<10.68          3   74705.04  14.11          2   15000.0+19070.0   7.51<5.47
Global *                    7  15000.0+59665.01    7.59<6.79          4   50127.04   9.47          3   15000.0+9537.07   5.41<2.73
Healthcare *                3  15000.0+26734.04    4.24<3.04          1     4312.0   0.81          2  15000.0+22422.04   8.24<6.43
Individual_Stocks           1          28693.08         2.92          0        0.0   0.00          1          28693.08        6.32
Large_Blend                 3          45011.04         4.58          2   23990.04   4.53          1           21021.0        4.63
Large_Growth                4         116004.02        11.80          2   83871.06  15.85          2          32132.06        7.08
Large_Value *               4  15000.0+48812.01    6.49<5.56          2   33788.07   6.38          2  15000.0+15023.04   6.61<4.31
Mid_Blend_Aggress           3          75099.03         7.64          0        0.0   0.00          3          75099.03       16.54
Mid_Growth                  5          91000.03         9.25          3   73643.03  13.91          2           17357.0        3.82
Mid_Value                   5           55137.0         5.61          3    31159.0   5.89          2           23978.0        5.28
REIT *                      6  15000.0+110356.0  12.75<12.57          4   77084.03  14.56          2  15000.0+33271.07  10.63<9.53
Small_Blend                 2          23400.07         2.38          1    4493.03   0.85          1          18907.04        4.17
Small_Growth *              5  30000.0+84902.02   11.69<9.67          3   69216.02  13.08          2   30000.0+15686.0  10.06<4.50
Small_Value                 2          19668.08         2.00          1     2920.0   0.55          1          16748.08        3.69
Totals                     55         983256.47       100.00         29  529308.36  99.99         26         453944.51       99.99
                                                                                                                                  

Processing:  --pt lp_diff

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                    Before                         After                        Before                        After                        Before                        After
Asset                Tot_%    Asset                Tot_%    Asset                Ret_%    Asset               Ret_%    Asset                Non_%    Asset               Non_%
Small_Value           2.24    Small_Value           2.24    Mid_Blend_Aggress     0.00    Mid_Blend_Aggress    0.00    Global                2.73    Global               2.73
Small_Blend           2.66    Small_Blend           2.66    Small_Value           0.55    Small_Value          0.55    Large_Value           4.31    Large_Value          4.31
Healthcare            3.04    Healthcare            3.04    Healthcare            0.81    Healthcare           0.81    Small_Growth          4.50    Small_Growth         4.50
Large_Blend           5.13    Large_Blend           5.13    Small_Blend           0.85    Small_Blend          0.85    Small_Value           4.80    Small_Value          4.80
Large_Value           5.56    Large_Value           5.56    Large_Blend           4.53    Large_Blend          4.53    Mid_Growth            4.97    Mid_Growth           4.97
Mid_Value             6.28    Mid_Value             6.28    Mid_Value             5.89    Mid_Value            5.89    Small_Blend           5.42    Small_Blend          5.42
Global                6.79    Global                6.79    Large_Value           6.38    Large_Value          6.38    Energy                5.46    Energy               5.46
Mid_Blend_Aggress     8.55    Mid_Blend_Aggress     8.55    Global                9.47    Global               9.47    Large_Blend           6.02    Large_Blend          6.02
Small_Growth          9.67    Small_Growth          9.67    Small_Growth         13.08    Small_Growth        13.08    Healthcare            6.43    Healthcare           6.43
Mid_Growth           10.36    Mid_Growth           10.36    Mid_Growth           13.91    Mid_Growth          13.91    Mid_Value             6.87    Mid_Value            6.87
Energy               10.68    Energy               10.68    Energy               14.11    Energy              14.11    Large_Growth          9.21    Large_Growth         9.21
REIT                 12.57    REIT                 12.57    REIT                 14.56    REIT                14.56    REIT                  9.53    REIT                 9.53
Large_Growth         13.21    Large_Growth         13.21    Large_Growth         15.85    Large_Growth        15.85    Mid_Blend_Aggress    21.52    Mid_Blend_Aggress   21.52
Individual_Stocks     3.27    Individual_Stocks     3.27    Individual_Stocks     0.00    Individual_Stocks    0.00    Individual_Stocks     8.22    Individual_Stocks    8.22
Totals              100.01                        100.01                         99.99                        99.99                         99.99                        99.99

