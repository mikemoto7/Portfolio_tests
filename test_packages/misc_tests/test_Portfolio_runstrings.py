import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






