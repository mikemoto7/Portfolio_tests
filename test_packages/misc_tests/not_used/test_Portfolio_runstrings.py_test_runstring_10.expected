
Processing: user_runstring = ./Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml

Processing: --price_web_update

Processing:  --pt show_percent

--------------------------------------------------------------------------------------------------------
                          Tot       Tot    Tot        Ret       Ret    Ret    Non_Ret   Non_Ret  Non_Ret
Asset_Class         inv_count       amt    pct  inv_count       amt    pct  inv_count       amt      pct
Energy                      4   93850.2  10.69          3   74764.2  14.14          1   19086.0     5.46
Global                      6   59391.2   6.76          4   49862.5   9.43          2    9528.7     2.73
Healthcare                  2   26725.1   3.04          1    4310.5   0.81          1   22414.6     6.42
Individual_Stocks           1   28803.6   3.28          0       0.0   0.00          1   28803.6     8.25
Large_Blend                 3   45069.9   5.13          2   24038.4   4.55          1   21031.5     6.02
Large_Growth                4  115412.1  13.14          2   83375.8  15.76          2   32036.3     9.17
Large_Value                 3   48781.8   5.55          2   33768.9   6.38          1   15012.9     4.30
Mid_Blend_Aggress           3   75211.4   8.56          0       0.0   0.00          3   75211.4    21.53
Mid_Growth                  5   90689.3  10.33          3   73413.1  13.88          2   17276.2     4.95
Mid_Value                   5   55138.6   6.28          3   31114.6   5.88          2   24024.0     6.88
REIT                        5  111152.0  12.66          4   77639.6  14.68          1   33512.4     9.59
Small_Blend                 2   23377.5   2.66          1    4466.7   0.84          1   18910.8     5.41
Small_Growth                4   84902.2   9.67          3   69216.2  13.09          1   15686.0     4.49
Small_Value                 2   19711.0   2.24          1    2927.0   0.55          1   16784.0     4.80
Totals                     49  878215.9  99.99         29  528897.5  99.99         20  349318.4   100.00
                                                                                                        

Processing:  --pt lp

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------
                    Sorted                                               Sorted                                               Sorted
Asset                Tot_%  Ret_%   Non_%     Asset               Tot_%   Ret_%   Non_%     Asset               Tot_%  Ret_%   Non_%
Small_Value           2.24   0.55    4.80     Mid_Blend_Aggress    8.56    0.00   21.53     Global               6.76   9.43    2.73
Small_Blend           2.66   0.84    5.41     Small_Value          2.24    0.55    4.80     Large_Value          5.55   6.38    4.30
Healthcare            3.04   0.81    6.42     Healthcare           3.04    0.81    6.42     Small_Growth         9.67  13.09    4.49
Large_Blend           5.13   4.55    6.02     Small_Blend          2.66    0.84    5.41     Small_Value          2.24   0.55    4.80
Large_Value           5.55   6.38    4.30     Large_Blend          5.13    4.55    6.02     Mid_Growth          10.33  13.88    4.95
Mid_Value             6.28   5.88    6.88     Mid_Value            6.28    5.88    6.88     Small_Blend          2.66   0.84    5.41
Global                6.76   9.43    2.73     Large_Value          5.55    6.38    4.30     Energy              10.69  14.14    5.46
Mid_Blend_Aggress     8.56   0.00   21.53     Global               6.76    9.43    2.73     Large_Blend          5.13   4.55    6.02
Small_Growth          9.67  13.09    4.49     Small_Growth         9.67   13.09    4.49     Healthcare           3.04   0.81    6.42
Mid_Growth           10.33  13.88    4.95     Mid_Growth          10.33   13.88    4.95     Mid_Value            6.28   5.88    6.88
Energy               10.69  14.14    5.46     Energy              10.69   14.14    5.46     Large_Growth        13.14  15.76    9.17
REIT                 12.66  14.68    9.59     REIT                12.66   14.68    9.59     REIT                12.66  14.68    9.59
Large_Growth         13.14  15.76    9.17     Large_Growth        13.14   15.76    9.17     Mid_Blend_Aggress    8.56   0.00   21.53
Individual_Stocks     3.28   0.00    8.25     Individual_Stocks    3.28    0.00    8.25     Individual_Stocks    3.28   0.00    8.25
Totals               99.99  99.99  100.00     Totals              99.99   99.99  100.00     Totals              99.99  99.99  100.00

Processing:  --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing: --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing:  --pt show_percent

----------------------------------------------------------------------------------------------------------------------------------
                          Tot               Tot          Tot        Ret        Ret    Ret    Non_Ret           Non_Ret     Non_Ret
Asset_Class         inv_count               amt          pct  inv_count        amt    pct  inv_count               amt         pct
Energy *                    5  15000.0+93850.02  11.07<10.69          3   74764.02  14.14          2   15000.0+19086.0   7.50<5.46
Global *                    7  15000.0+59391.02    7.57<6.76          4   49862.05   9.43          3   15000.0+9528.07   5.40<2.73
Healthcare *                3  15000.0+26725.01    4.24<3.04          1    4310.05   0.81          2  15000.0+22414.06   8.24<6.42
Individual_Stocks           1          28803.06         2.93          0        0.0   0.00          1          28803.06        6.34
Large_Blend                 3          45069.09         4.58          2   24038.04   4.54          1          21031.05        4.63
Large_Growth                4         115412.01        11.74          2   83375.08  15.76          2          32036.03        7.05
Large_Value *               4  15000.0+48781.08    6.49<5.55          2   33768.09   6.38          2  15000.0+15012.09   6.61<4.30
Mid_Blend_Aggress           3          75211.04         7.65          0        0.0   0.00          3          75211.04       16.55
Mid_Growth                  5          90689.03         9.22          3   73413.01  13.88          2          17276.02        3.80
Mid_Value                   5          55138.06         5.61          3   31114.06   5.88          2           24024.0        5.29
REIT *                      6  15000.0+111152.0  12.83<12.66          4   77639.06  14.68          2  15000.0+33512.04  10.68<9.59
Small_Blend                 2          23377.05         2.38          1    4466.07   0.84          1          18910.08        4.16
Small_Growth *              5  30000.0+84902.02   11.69<9.67          3   69216.02  13.09          2   30000.0+15686.0  10.06<4.49
Small_Value                 2           19711.0         2.00          1     2927.0   0.55          1           16784.0        3.69
Totals                     55         983211.49       100.00         29  528892.55  99.98         26         454313.54      100.00
                                                                                                                                  

Processing:  --pt lp_diff

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                    Before                        After                        Before                        After                        Before                         After
Asset                Tot_%    Asset               Tot_%    Asset                Ret_%    Asset               Ret_%    Asset                Non_%    Asset                Non_%
Small_Value           2.24    Small_Value          2.24    Mid_Blend_Aggress     0.00    Mid_Blend_Aggress    0.00    Global                2.73    Global                2.73
Small_Blend           2.66    Small_Blend          2.66    Small_Value           0.55    Small_Value          0.55    Large_Value           4.30    Large_Value           4.30
Healthcare            3.04    Healthcare           3.04    Healthcare            0.81    Healthcare           0.81    Small_Growth          4.49    Small_Growth          4.49
Large_Blend           5.13    Large_Blend          5.13    Small_Blend           0.84    Small_Blend          0.84    Small_Value           4.80    Small_Value           4.80
Large_Value           5.55    Large_Value          5.55    Large_Blend           4.55    Large_Blend          4.55    Mid_Growth            4.95    Mid_Growth            4.95
Mid_Value             6.28    Mid_Value            6.28    Mid_Value             5.88    Mid_Value            5.88    Small_Blend           5.41    Small_Blend           5.41
Global                6.76    Global               6.76    Large_Value           6.38    Large_Value          6.38    Energy                5.46    Energy                5.46
Mid_Blend_Aggress     8.56    Mid_Blend_Aggress    8.56    Global                9.43    Global               9.43    Large_Blend           6.02    Large_Blend           6.02
Small_Growth          9.67    Small_Growth         9.67    Small_Growth         13.09    Small_Growth        13.09    Healthcare            6.42    Healthcare            6.42
Mid_Growth           10.33    Mid_Growth          10.33    Mid_Growth           13.88    Mid_Growth          13.88    Mid_Value             6.88    Mid_Value             6.88
Energy               10.69    Energy              10.69    Energy               14.14    Energy              14.14    Large_Growth          9.17    Large_Growth          9.17
REIT                 12.66    REIT                12.66    REIT                 14.68    REIT                14.68    REIT                  9.59    REIT                  9.59
Large_Growth         13.14    Large_Growth        13.14    Large_Growth         15.76    Large_Growth        15.76    Mid_Blend_Aggress    21.53    Mid_Blend_Aggress    21.53
Individual_Stocks     3.28    Individual_Stocks    3.28    Individual_Stocks     0.00    Individual_Stocks    0.00    Individual_Stocks     8.25    Individual_Stocks     8.25
Totals               99.99                        99.99                         99.99                        99.99                        100.00                        100.00

